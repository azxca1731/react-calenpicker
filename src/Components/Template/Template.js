import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const TemplateDiv = styled.div`
  width: 300px;
  float: left;
  background-color: ${props => props.theme.backgroundColor};
  border: 1px solid ${props => props.theme.borderColor};
`;

const Devider = styled.div`
  width: 100%;
  border-bottom: 1px solid ${props => props.theme.borderColor};
`;

const TemplateLight = props => {
  const { head, children, cssObject } = props;
  return (
    <TemplateDiv style={cssObject}>
      {head}
      <Devider />
      {children}
    </TemplateDiv>
  );
};

TemplateLight.defaultProps = {
  cssObject: {}
};

TemplateLight.propTypes = {
  head: PropTypes.element.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
  cssObject: PropTypes.object
};
export default TemplateLight;
