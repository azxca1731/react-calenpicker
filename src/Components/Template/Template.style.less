.Template {
  width: 300px;
  float: left;
}
