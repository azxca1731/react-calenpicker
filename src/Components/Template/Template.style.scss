.Template {
  width: 300px;
  float: left;
  background-color: $background-color;
  border: 1px solid $border-color;
  &__devider {
    width: 100%;
    border-bottom: 1px solid $border-color;
  }
}
