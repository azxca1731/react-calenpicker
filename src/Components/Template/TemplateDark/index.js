import TemplateDark from "./TemplateDark";

export default TemplateDark;
