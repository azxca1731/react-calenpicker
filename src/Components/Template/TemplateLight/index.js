import TemplateLight from "./TemplateLight";

export default TemplateLight;
