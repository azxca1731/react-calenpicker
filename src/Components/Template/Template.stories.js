import React from "react";

import { storiesOf } from "@storybook/react";

import Template from "Components/Template";

storiesOf("Components", module).add("Template", () => <Template />);
