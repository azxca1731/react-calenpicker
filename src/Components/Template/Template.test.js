import React from "react";
import { shallow } from "enzyme";
import Template from "index";

describe("Template", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<Template />);
    expect(wrapper).toHaveLength(1);
  });
});
