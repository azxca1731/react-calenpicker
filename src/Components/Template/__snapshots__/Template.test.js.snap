// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template render custom css render well with snapshot 1`] = `
<styled.div
  duplicated={false}
  style={
    Object {
      "color": "red",
    }
  }
>
  <div />
  <styled.div />
  <div />
  <AddButton
    handleModal={[Function]}
  />
</styled.div>
`;

exports[`Template renders with snapshot 1`] = `
<styled.div
  duplicated={false}
  style={Object {}}
>
  <div />
  <styled.div />
  <div />
  <AddButton
    handleModal={[Function]}
  />
</styled.div>
`;
